libraries/lib-components/EffectInterface.cpp

```diff
@@ -160,4 +160,21 @@ EffectDefinitionInterfaceEx::FindMe(const Settings &settings) const
 
 EffectProcessor::~EffectProcessor() = default;
 
+EffectUIValidator::~EffectUIValidator() = default;
+
+DefaultEffectUIValidator::DefaultEffectUIValidator(
+   EffectUIClientInterface &effect)
+   : mEffect{effect}
+{}
+
+DefaultEffectUIValidator::~DefaultEffectUIValidator()
+{
+   mEffect.CloseUI();
```

Comment:
I am guessing that the purpose of this class is to have something which we can query to check whether all numeric values in the text fields of the UI of an effect are in the valid range (I count only text fields - because I assume that the possible range of motion of knobs and sliders is already bounded to a valid range).

What I'd not expect here, is that when this validation object gets destroyed, the User Interface is closed too (as in, the dialog will disappear). Are not these two things unrelated?

Reply:
You are correct about the purpose of the sole virtual function of the class.

See the existing overrides of EffectUIClientInterface::CloseUI() which still had to be called somewhere after this reorganization.  It does not cause the dialog to disappear -- rather, it happens during `EffectUIHost::OnClose` which is preparing the dialog for destruction.

It is a place where certain cleanup tasks must happen (like popping the event handler on a window), complementary to the setup tasks that are done in `EffectUIClientInterface::PopulateUI` which is the factory for this object.

The object exists to do more than satisfy the virtual function interface:  it also sets itself up to handle user interface events, and update the EffectSettings object that is accessible through the `access` argument that was given to `PopulateUI`.